crates/matrix-sdk/src/room/common.rs

```diff
@@ -817,6 +819,56 @@ impl Common {
         let request = delete_tag::v3::Request::new(&user_id, self.inner.room_id(), tag.as_ref());
         self.client.send(request, None).await
     }
+
+    /// Sets whether this room is a DM.
+    ///
+    /// When setting this room as DM, it will be marked as DM for all active
+    /// members of the room. When unsetting this room as DM, it will be
+    /// unmarked as DM for all users, not just the members.
+    ///
+    /// # Arguments
+    /// * `is_direct` - Whether to mark this room as direct.
+    pub async fn set_is_direct(&self, is_direct: bool) -> Result<()> {
+        let user_id = self
+            .client
+            .user_id()
+            .await
+            .ok_or_else(|| Error::from(HttpError::AuthenticationRequired))?;
+
+        let mut content = self
+            .client
+            .store()
+            .get_account_data_event(GlobalAccountDataEventType::Direct)
```

Comment:
Do we not have a `_static` variant of this method?

Reply:
what do you mean?

Reply:
We have `get_foo_event_static` in a few places that you could call as `get_account_data_event_static::<DirectEvent>()` here, if it existed (but it doesn't). Can be done in a future refactoring.

Reply:
oh i didn't see that.